### PR TITLE
fix(storage): passes the uploaded size

### PR DESCRIPTION
**Pull request by jurajpiar**

### CODE REVIEW

File: src/api/rif-storage-upload-service/upload/index.ts [hunk 1]
```diff
@@ -67,8 +70,8 @@ export default class UploadService
           id: 'service-post',
         })
       }
-      const data = await response.json()
+      const data: UploadResponse = await response.json()
 
-      return data.fileHash
+      return data
```
Comment (farinalvaro) [inline src/api/rif-storage-upload-service/upload/index.ts:75]: don't need to await here as the return type is already a Promise
```suggestion
      return response.json()
```

File: src/api/rif-storage-upload-service/upload/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Big } from 'big.js'
+import { APIService } from 'api/models/apiService'
+import { Modify } from 'utils/typeUtils'
+
+export type UploadResponse = {
+    fileHash: string
+    fileSize: number
```
Comment (farinalvaro) [inline src/api/rif-storage-upload-service/upload/interfaces.ts:7]: shouldn't size be `Big` type?

File: src/api/rif-storage-upload-service/upload/interfaces.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    message: string
+}
+
+export type FileSizeResponse = {
+    fileHash: string
+    fileSizeBytes: number
```
Comment (farinalvaro) [inline src/api/rif-storage-upload-service/upload/interfaces.ts:13]: shouldn't size be Big type?

File: src/components/molecules/storage/upload/StorageUploaded.tsx [hunk 4]
```diff
@@ -28,7 +29,7 @@ const StorageUploaded: FC<Props> = ({
       <Typography variant="subtitle1" color="secondary">The hash of your upload is:</Typography>
     </GridItem>
     <GridItem>
-      <Typography variant="caption" color="secondary">{uploadedHash}</Typography>
+      <Typography variant="caption" color="secondary">{uploadResponse?.fileHash}</Typography>
     </GridItem>
```
Comment (farinalvaro) [inline src/components/molecules/storage/upload/StorageUploaded.tsx:33]: Should we render this `GridItem` if uploadResponse is undefined?

File: src/components/organisms/storage/buy/PinningCard.tsx [hunk 5]
```diff
@@ -128,13 +132,16 @@ const PinningCard: FC<Props> = ({ dispatch }) => {
...
-          && parseConvertBig(sizeB.minus(TOTAL_SIZE_LIMIT),
-            UNIT_PREFIX_POW2.MEGA).toFixed(3)}
+        sizeOverLimitMB={uploadDisabled
+          && hasSize
+          && parseConvertBig(
+            size.minus(TOTAL_SIZE_LIMIT),
+            UNIT_PREFIX_POW2.MEGA,
+          ).toFixed(3)}
         maxSizeMB={parseConvertBig(TOTAL_SIZE_LIMIT,
           UNIT_PREFIX_POW2.MEGA).toString()}
```
Comment (farinalvaro) [inline src/components/organisms/storage/buy/PinningCard.tsx:146]: Would be clearer if extracted to constant I think

File: src/components/organisms/storage/buy/PinningCard.tsx [hunk 6]
```diff
@@ -187,15 +192,13 @@ const PinningCard: FC<Props> = ({ dispatch }) => {
...
               ) => acc + file.size, 0))
-              setSizeB(totalB)
-              const sizeUnit = totalB.div(unit)
+              setSize(totalB)
               setUploadDisabled(!!addedFiles.length
                 && totalB.gt(TOTAL_SIZE_LIMIT))
-              setSize(sizeUnit.toString())
               setFiles(addedFiles)
             }}
             filesLimit={666 * 666 * 666}
```
Comment (farinalvaro) [inline src/components/organisms/storage/buy/PinningCard.tsx:200]: not related to this PR but what is this value? 😈

File: src/components/organisms/storage/buy/PinningCard.tsx [hunk 7]
```diff
@@ -187,15 +192,13 @@ const PinningCard: FC<Props> = ({ dispatch }) => {
               const totalB = new Big(addedFiles.reduce((
                 acc, file,
               ) => acc + file.size, 0))
-              setSizeB(totalB)
-              const sizeUnit = totalB.div(unit)
+              setSize(totalB)
               setUploadDisabled(!!addedFiles.length
                 && totalB.gt(TOTAL_SIZE_LIMIT))
-              setSize(sizeUnit.toString())
               setFiles(addedFiles)
```
Comment (farinalvaro) [inline src/components/organisms/storage/buy/PinningCard.tsx:198]: I think this function could be cleaner if extracted

**Review by farinalvaro — CHANGES_REQUESTED** [review]
Nice! Just left some suggestions and a couple of questions.

File: src/components/organisms/storage/buy/PinningCard.tsx (hunk 7 above)
Comment (jurajpiar) [reply src/components/organisms/storage/buy/PinningCard.tsx:198]: Disagree

File: src/components/organisms/storage/buy/PinningCard.tsx (hunk 6 above)
Comment (jurajpiar) [reply src/components/organisms/storage/buy/PinningCard.tsx:200]: 295,408,296

File: src/components/molecules/storage/upload/StorageUploaded.tsx (hunk 4 above)
Comment (jurajpiar) [reply src/components/molecules/storage/upload/StorageUploaded.tsx:33]: No it's always defined in runtime I think

File: src/api/rif-storage-upload-service/upload/interfaces.ts (hunk 2 above)
Comment (jurajpiar) [reply src/api/rif-storage-upload-service/upload/interfaces.ts:7]: Nope

File: src/api/rif-storage-upload-service/upload/interfaces.ts (hunk 3 above)
Comment (jurajpiar) [reply src/api/rif-storage-upload-service/upload/interfaces.ts:13]: Nope

File: src/api/rif-storage-upload-service/upload/index.ts (hunk 1 above)
Comment (jurajpiar) [reply src/api/rif-storage-upload-service/upload/index.ts:75]: No actually just forgot to map it

File: src/components/organisms/storage/buy/PinningCard.tsx (hunk 6 above)
Comment (jurajpiar) [reply src/components/organisms/storage/buy/PinningCard.tsx:200]: If that's not sufficiently large I can do Big(6).pow(6).pow(6) instead .]

Comment (jurajpiar) [reply src/components/organisms/storage/buy/PinningCard.tsx:200]: should you want to know more about this number: https://numbermatics.com/n/295408296/

File: src/components/molecules/storage/upload/StorageUploaded.tsx (hunk 4 above)
Comment (jurajpiar) [reply src/components/molecules/storage/upload/StorageUploaded.tsx:33]: Adapted the component disallow optional uploadResponse

File: src/api/rif-storage-upload-service/upload/interfaces.ts (hunk 2 above)
Comment (jurajpiar) [reply src/api/rif-storage-upload-service/upload/interfaces.ts:7]: no we receive and process it as Number

File: src/api/rif-storage-upload-service/upload/interfaces.ts (hunk 3 above)
Comment (jurajpiar) [reply src/api/rif-storage-upload-service/upload/interfaces.ts:13]: no we receive and process it as Number

Comment (jurajpiar) [reply src/api/rif-storage-upload-service/upload/interfaces.ts:13]: no we receive and process it as Number

**Review by farinalvaro — APPROVED** [review]
👌

**Pull request merged**